### ISSUE TITLE
Could I build custom optimizer and lr scheduler with plugin way?

### ISSUE THREAD

**Issue by BaohaoLiao** [pytorch/fairseq#3811]
#### What is your question?
I want to build a new optimizer and lr scheduler with the plugin way, i.e. using the flag --user-dir. I could build custom model, criterion and task. However, when I build optimizer and lr scheduler, It doesn't work. It always show it is an invalid choice.  Is it possible for us to build a custom optimizer or lr scheduler in a plugin way? I could not find this kind of example in the source code.

#### Code
My user dir looks like:
  |----my_dir
  |-------- criterions
  |------------__init__.py
  |------------my_criterion.py
  |--------models
  |------------__init__.py
  |------------my_model.py
  |--------optim
  |------------__init__.py
  |------------my_optim.py
  |------------lr_scheduler
  |----------------__init__.py
  |----------------my_lr_scheduler.py

Only the lr scheduler and optimizer do not work. 

#### What's your environment?

 - fairseq Version (e.g., 1.0 or master): 0.10.2
 - PyTorch Version (e.g., 1.0): 1.8
 - OS (e.g., Linux): Linux
 - How you installed fairseq (`pip`, source): pip
 - Python version: 3.7
 - CUDA/cuDNN version: 10.1
